pwndbg/commands/heap.py

```diff
@@ -166,23 +122,112 @@ def malloc_chunk(addr):
     """
     Prints out the malloc_chunk at the specified address.
     """
+    main_heap = pwndbg.heap.get_heap()
+
     if not isinstance(addr, six.integer_types):
         addr = int(addr)
 
     chunk = value_from_type('struct malloc_chunk', addr)
     size = int(chunk['size'])
-    prev_inuse = (size & PREV_INUSE) == 1
-    is_mmaped = (size & IS_MMAPED) == 1
-    non_main_arena = (size & NON_MAIN_ARENA) == 1
+
+    prev_inuse, is_mmapped, non_main_arena = main_heap.chunk_flags(size)
 
     header = M.get(addr)
     if prev_inuse:
```

Comment:
![heap_fastbins](https://cloud.githubusercontent.com/assets/20363764/21030467/7c370f1c-bd53-11e6-8127-3121b2f0eede.png)

I implemented a check in the`malloc_chunk` command. If it is a fastbin then display that instead of PREV_INUSE. To figure out if the previous heap chunk is in use when a fastbin is after it, I believe we can go up the heap until we see the next chunk that isn't a fastbin.